pkg/reconciler/mongodb/mongodbsource.go

```diff
@@ -14,4 +14,155 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package mongodb
+// Code generated by injection-gen. DO NOT EDIT.
+
+package mongodbsource
+
+import (
+	"context"
+	// "errors"
+	"fmt"
+	"go.uber.org/zap"
+	"knative.dev/pkg/logging"
+
+	"k8s.io/client-go/kubernetes"
+
+	v1alpha1 "github.com/googleinterns/knative-source-mongodb/pkg/apis/sources/v1alpha1"
+	mongodbsource "github.com/googleinterns/knative-source-mongodb/pkg/client/injection/reconciler/sources/v1alpha1/mongodbsource"
+	// "go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.mongodb.org/mongo-driver/mongo/readpref"
+	v1 "k8s.io/api/core/v1"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	reconciler "knative.dev/pkg/reconciler"
+)
+
+// TODO: PLEASE COPY AND MODIFY THIS FILE AS A STARTING POINT
+
+// newReconciledNormal makes a new reconciler event with event type Normal, and
+// reason MongoDbSourceReconciled.
+func newReconciledNormal(namespace, name string) reconciler.Event {
+	return reconciler.NewEvent(v1.EventTypeNormal, "MongoDbSourceReconciled", "MongoDbSource reconciled: \"%s/%s\"", namespace, name)
+}
+
+// Reconciler implements controller.Reconciler for MongoDbSource resources.
+type Reconciler struct {
+	// Clients
+	kubeClientSet kubernetes.Interface
+
+	// Lister
+	secretLister corev1listers.SecretLister
+}
+
+// Check that our Reconciler implements Interface
+var _ mongodbsource.Interface = (*Reconciler)(nil)
+
+// Optionally check that our Reconciler implements Finalizer
+//var _ mongodbsource.Finalizer = (*Reconciler)(nil)
+
+// Optionally check that our Reconciler implements ReadOnlyInterface
+// Implement this to observe resources even when we are not the leader.
+//var _ mongodbsource.ReadOnlyInterface = (*Reconciler)(nil)
+
+// Optionally check that our Reconciler implements ReadOnlyFinalizer
+// Implement this to observe tombstoned resources even when we are not
+// the leader (best effort).
+//var _ mongodbsource.ReadOnlyFinalizer = (*Reconciler)(nil)
+
+// ReconcileKind implements Interface.ReconcileKind.
+func (r *Reconciler) ReconcileKind(ctx context.Context, src *v1alpha1.MongoDbSource) reconciler.Event {
+	src.Status.InitializeConditions()
+	src.Status.ObservedGeneration = src.Generation
+
+	// Try to connect to db and see if it works
+	fmt.Printf("----------------------------------------------------------------------------------------------------------------------------------")
+	fmt.Printf(src.Spec.Secret.Name)
+
+	secret, err := r.secretLister.Secrets(src.Namespace).Get(src.Spec.Secret.Name)
+	if err != nil {
+		logging.FromContext(ctx).Error("Unable to read MongoDb credentials secret", zap.Error(err))
+		return err
+	}
+	rawURI, ok := secret.Data["URI"]
```

Comment:
nit: I think I'd rather call the key "connectionStr" instead of URI. But I don't have a strong preference

Reply:
I just called it URI because the mongodb function is called `mongo.NewClient(options.Client().ApplyURI(_____)` 
Let me know if you still want to change it, I can do that too!

Reply:
sounds good, let's leave it as URI then, thanks!